src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,154 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private T data;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T data, Node<T> next) {
+            this.prev = prev;
+            this.data = data;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            Node<T> current = new Node<>(tail, value, null);
+            tail.next = current;
+            tail = current;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+        if (size == 0 || index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node current = new Node<>(null, value, head);
+            head.prev = current;
+            head = current;
+            size++;
+        } else {
+            Node<T> prev = getNodeByIndex(index - 1);
+            Node<T> next = prev.next;
+            Node<T> current = new Node<>(prev, value, next);
+            prev.next = current;
+            next.prev = current;
+            size++;
+        }
+
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T el : list) {
+            add(el);
+        }
     }
 
     @Override
     public T get(int index) {
```

Comment:
I don't think you should create another method for this, it's hard to read

Reply:
This is a part of task